it/uk/gov/hmrc/fhregistrationfrontend/controllers/BusinessPartnersUnincorporatedBodyNameControllerISpec.scala

```diff
@@ -1,90 +1,138 @@
+  def route(mode: Mode): String = routes.BusinessPartnersUnincorporatedBodyNameController.load(index = 1, mode).url.drop(6)
 
-  s"GET $route" should {
-    "render the Unincorporated Body Name page" in {
-      given
-        .commonPrecondition
+  val userAnswersWithPageData = emptyUserAnswers.set[UnincorporatedBodyName](UnincorporatedBodyNamePage(1), UnincorporatedBodyName("company name"))
+    .success
+    .value
 
-      val result = buildRequest(route)
-        .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
-        .get()
+  List(NormalMode, CheckMode).foreach { mode =>
+    s"GET ${route(mode)}" should {
+      "render the Unincorporated Body Name page" when {
+        "there is no form data" in {
+          given
+            .commonPrecondition
+          addUserAnswersToSession(emptyUserAnswers)
 
-      whenReady(result) { res =>
-        res.status mustBe 200
-        val page = Jsoup.parse(res.body)
-        page.title() must include("What is the unincorporated body name?")
-        page.getElementsByTag("h1").text() must include("What is the unincorporated body name?")
+          val result = buildRequest(route(mode))
+            .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
+            .get()
+
+          whenReady(result) { res =>
+            res.status mustBe 200
+            val page = Jsoup.parse(res.body)
+            page.title() must include("What is the unincorporated body name?")
+            page.getElementsByTag("h1").text() must include("What is the unincorporated body name?")
+          }
+        }
       }
-    }
-  }
 
-  s"POST $route" when {
-    "the unincorporated body name is entered" should {
-      "redirect to the Unincorporated Body Trading Name page" in {
+      "there is form data" in {
         given
           .commonPrecondition
+        addUserAnswersToSession(userAnswersWithPageData)
 
-        val result = buildRequest(route)
+        val result = buildRequest(route(mode))
           .addCookies(DefaultWSCookie("mdtp", authAndSessionCookie))
-          .withHttpHeaders(xSessionId,
-            "Csrf-Token" -> "nocheck")
-          .post(Map(
-            "unincorporatedBodyName_value" -> Seq("Test Body")
-          ))
+          .get()
 
         whenReady(result) { res =>
-          res.status mustBe 303
-          res.header(HeaderNames.LOCATION) mustBe Some(unincorpBodyTradingNameUrl)
+          res.status mustBe 200
+          val page = Jsoup.parse(res.body)
+          page.title() must include("What is the unincorporated body name?")
+          page.getElementsByTag("h1").text() must include("What is the unincorporated body name?")
+          val companyName = page.getElementById("unincorporatedBodyName_value")
+          companyName.attr("value") mustBe "company name"
+        }
+      }
+
+      "redirect the user to the start of the BusinessPartners journey" when {
```

Comment:
The user is redirected to the start of the FHDDS journey now not the businessPartners journey

Reply:
Changes made